src/isar/state_machine/states/monitor.py

```diff
@@ -29,228 +16,14 @@ class Monitor(State):
     @inject
     def __init__(self, state_machine: "StateMachine") -> None:
         super().__init__(name="monitor", on_enter=self.start, on_exit=self.stop)
-        self.state_machine: "StateMachine" = state_machine
 
-        self.logger = logging.getLogger("state_machine")
-        self.events = self.state_machine.events
-
-        self.awaiting_task_status: bool = False
-
-        self.signal_state_machine_to_stop: Event = (
-            state_machine.signal_state_machine_to_stop
+        self.generic_state = OngoingMission(
+            state_machine=state_machine,
+            state=OngoingMissionStates.Monitor,
```

Comment:
Not sure if it is better or even possible, but perhaps the Monitor and ReturningHome states could inherit the OngoingMission class:
`class Monitor(State):`
This way we would not have to call `monitor.generic_state.state_machine.current_mission` but could call `monitor.state_machine.current_mission` directly